explore/src/explore.cpp

```diff
@@ -396,11 +427,13 @@ void Explore::start()
 void Explore::stop(bool finished_exploring)
 {
   RCLCPP_INFO(logger_, "Exploration stopped.");
-  move_base_client_->async_cancel_all_goals();
   exploring_timer_->cancel();
 
-  if (return_to_init_ && finished_exploring) {
-    returnToInitialPose();
+  if (move_base_control_ && return_to_init_ && finished_exploring) {
+    RCLCPP_INFO(logger_, "Returning to initial pose.");
+    move_base_client_->async_cancel_all_goals();
+    sendGoalToMoveBase(initial_pose_.position, initial_pose_.orientation,
+                       false);
   }
```

Comment:
This stop function can be called even if `return_to_init =False` and `finished_exploring=False`
Please add a separate conditional to check `move_base_control` and cancel the goals